### PR TITLE
chore: update to latest Redis (aioredis is deprecated)

### PR DESCRIPTION
**Pull request by kentbull**
As of Feb 21, 2023 aioredis-py was archived. See the package repo here:
https://github.com/aio-libs-abandoned/aioredis-py

# Summary of Changes

A docs update to correct the ASGI doc to use the latest version of Redis and swap the `aioredis` package for the `redis` package since `aioredis` is now included within `redis`.

# Pull Request Checklist

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [ ] Applied changes to both WSGI and ASGI code paths and interfaces (where applicable).
- [X ] Added **tests** for changed code.
- [ X] Prefixed code comments with GitHub nick and an appropriate prefix.
- [ X] Coding style is consistent with the rest of the framework.
- [ X] Updated **documentation** for changed code.
    - [X ] Added docstrings for any new classes, functions, or modules.
    - [ X] Updated docstrings for any modifications to existing code.
    - [ ] Updated both WSGI and ASGI docs (where applicable).
    - [ X] Added references to new classes, functions, or modules to the relevant RST file under `docs/`.
    - [ ] Updated all relevant supporting documentation files under `docs/`.
    - [ ] A copyright notice is included at the top of any new modules (using your own name or the name of your organization).
    - [ ] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/directives.html?highlight=versionadded#directive-versionadded).
- [ ] Changes (and possible deprecations) have [towncrier](https://towncrier.readthedocs.io/en/latest/quickstart.html#creating-news-fragments) news fragments under `docs/_newsfragments/`, with the file name format `{issue_number}.{fragment_type}.rst`. (Run `towncrier --draft` to ensure it renders correctly.)

### CODE REVIEW

File: docs/user/tutorial-asgi.rst [hunk 1]
```diff
@@ -747,9 +747,8 @@ implementations for production and testing.
     ``self.redis_host``. Such a design might prove helpful for apps that
```
Comment (vytas7) [inline docs/user/tutorial-asgi.rst:747]: This note might need to be updated or even removed altogether if it is no longer needed. How can we hook in `fakeredis` for testing then though? (Monkey-patching or mocking might work too.)

**Review by vytas7 — CHANGES_REQUESTED** [review]
Thanks for this [draft] PR!
It looks good for the most of it, just it seems that the CI gate for the tutorial is failing. And one note might need updating (depending on the path that we choose).

You can test the same gate locally as (I've just verified it still works on CPython 3.10):
```
tox -e asgilook
```

File: docs/user/tutorial-asgi.rst (hunk 1 above)
Comment (kentbull) [reply docs/user/tutorial-asgi.rst:747]: with the latest update I made the `redis_from_url` function is still used. I added a wrapper function to `config.py` to change as little as possible. Monkey patching still works.

**Comment by kentbull** [comment]
Tests pass, though coverage is not at 100%.

**Review by CaselIT — DISMISSED** [review]

**Review by vytas7 — APPROVED** [review]
I've tweaked your code slightly @kentbull, and retested it manually. Now everything seems to pass and work for me.

**Pull request merged**